spec/loaders/abstract-loader-shared-behavior.coffee

```diff
@@ -333,6 +356,23 @@ registerSharedBehavior "AbstractLoaderSharedBehavior", (sharedContext) ->
               loader.setup(opts)
               notFound(loader, opts)
 
+          context 'all of the cached models have their optional fields loaded', ->
+            beforeEach ->
+              taskOne.set('test_field', 'test value')
+
+            it 'calls #_onLoadSuccess with the models from the cache', ->
+              opts.optionalFields = ['test_field']
+              loader.setup(opts)
+              loader._checkCacheForData()
+              expect(loader._onLoadSuccess).toHaveBeenCalledWith([taskOne])
+
+          context 'all of the cached models do not have their optional fields loaded', ->
+            it 'returns false and does not call #_onLoadSuccess', ->
```

Comment:
Where do we test the return value?  Should that be here and in the other `it` block?